src/main/java/io/debezium/connector/vitess/VitessMetadata.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.vitess;
+
+import static java.lang.Math.toIntExact;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.ByteString;
+
+import io.debezium.annotation.VisibleForTesting;
+import io.debezium.connector.vitess.connection.VitessReplicationConnection;
+import io.vitess.proto.Query;
+import io.vitess.proto.Vtgate;
+
+/**
+ * Class for getting metadata on Vitess, e.g., tables, shards. Supports shard-specific queries.
+ */
+public class VitessMetadata {
```

Comment:
Would it be possible to convert this class to be no-static? A lot of methods uses `VitessConnectorConfig` so maybe an instance can be created that will receive the config in constructor.